monkey/infection_monkey/monkey.py

```diff
@@ -233,7 +236,8 @@ def start(self):
                     if not self._keep_running:
                         break
 
-                RansomewarePayload(WormConfiguration.ransomware).run_payload()
+                telemetry_messenger = TelemetryMessengerWrapper()
+                RansomewarePayload(WormConfiguration.ransomware, telemetry_messenger).run_payload()
```

Comment:
What's the point of initializing the messenger and passing it as a param if it doesn't need any properties of the current context? Just init it in the payload?

Reply:
@VakarisZ, @shreyamalviya Read [this](https://www.freecodecamp.org/news/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f/) for a quick intro to dependency injection or [this](https://martinfowler.com/articles/injection.html) for a more in-depth look. Also check out the "Inheritance Tax" section of The Pragmatic Programmer.

The short answer here is that, in a world where payloads are plugins, the mechanism for sending telemetry is subject to change. Providing the payload with something that's responsible for sending telemetry means we can change the the way telemetry is sent without affecting any of our payloads. Moreover, we can change the way telemetry is sent for different plugins, so telemetry can be sent in a variety of ways. To do this, we need to [remove the responsibility of sending telemetries from the BaseTelem class](https://github.com/guardicore/monkey/blob/6744ee71fc102128fe0c18c627d73ee059da78f0/monkey/infection_monkey/telemetry/base_telem.py#L13-L24) (but not today).